.github/workflows/static.yml

```diff
@@ -0,0 +1,56 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy static content to Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["master"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  generate-and-deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install dependencies and generate web documents
+        run: |
+          cd burrito_converter
+          python3 -m venv venv
+          source venv/bin/activate
+          pip install -r requirements.txt
+          mkdir web_docs
+          cd generators
+          source ../venv/bin/activate
+          python3 main.py --documentation
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v5
+
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: 'burrito_converter/web_docs'
```

Comment:
What is the difference between uploading and deploying github action pages?

Reply:
This action takes the files and turns them into a single file that the next action uses. It is a `.gzip` archive with a single `tar` file.

Reply:
Where does it get uploaded to?
and what happens if we do not run the deploy?

Reply:
:white_check_mark: